routes/api/events.js

```diff
@@ -1,9 +1,376 @@
+      website,
+      image,
+      date,
+    } = req.body;
+
+    // Build event object
+    const eventFields = {};
+    eventFields.user = req.user.id;
+    if (title) eventFields.title = title;
+    if (description) eventFields.description = description;
+    if (dateStart) eventFields.dateStart = dateStart;
+    if (dateEnd) eventFields.dateEnd = dateEnd;
+    if (type) eventFields.type = type;
+    if (location) eventFields.location = location;
+    if (website) eventFields.website = website;
+    if (image) eventFields.image = image;
+    if (date) eventFields.date = date;
+
+    try {
+      const event = await Event.updateOne(
+        {
+          user: req.user.id,
+          _id: req.params.id,
+        },
+        {
+          $set: {
+            title,
+            description,
+            dateStart,
+            dateEnd,
+            type,
+            location,
+          },
+        }
+      );
+
+      res.json(event);
+    } catch (err) {
+      console.error(err.message);
+      res.status(500).send('Server Error');
+    }
+  }
+);
+
+// @route   POST api/events/:id
+// @desc    Registration on event
+// @access  Public
+router.post(
+  '/id/:id',
+  [
+    check('name', 'Name is required').notEmpty(),
+    check('phone', 'Phone number is required').notEmpty(),
+  ],
+  async (req, res) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(400).json({ errors: errors.array() });
+    }
+
+    // Create unique code for participant
+    const shortId = shortid.generate();
+
+    const newParticipant = {
+      name: req.body.name,
+      phone: req.body.phone,
+      shortId,
+    };
+
+    try {
+      const event = await Event.findById(req.params.id);
+      const participant = await Event.find({
+        _id: req.params.id,
+        'participants.phone': req.body.phone,
+      });
+
+      // Check if participant exists
+      if (participant.length > 0) {
+        return res.status(200).send('You are already registred');
+      }
+
+      // Add participant to collection and save
+      event.participants.push(newParticipant);
+      await event.save();
+
+      res.status(200).send(shortId);
+    } catch (err) {
+      console.error(err.message);
+      res.status(500).send('Server Error');
+    }
+  }
+);
+
+// @route   POST api/events/status
+// @desc    Check registration status
 // @access  Public
-router.get('/', (req, res) => res.send('Events route'));
+router.post(
+  '/status',
+  [check('shortId', 'Registration code is required').notEmpty()],
+  async (req, res) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(400).json({ errors: errors.array() });
+    }
+
+    try {
+      const event = await Event.find({
+        'participants.shortId': req.body.shortId,
+      });
+
+      // Check if participant exists
+      if (event.length === 0) {
+        return res.status(404).send('Wrong registration code');

```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/spumony/xvent-project-backend/pullRequest?prid=7531905)
```suggestion
        return res.status(404).send("Wrong registration code");
```